pkg/kyamls/helpers.go

```diff
@@ -30,41 +30,35 @@ func GetNamespace(node *yaml.RNode, path string) string {
 	return GetStringField(node, path, "metadata", "namespace")
 }
 
-func getMetadataMap(node *yaml.RNode, path string, mapName string) (map[string]string, error) {
-	metadata, err := node.Pipe(yaml.Lookup("metadata", mapName))
+// GetMap return the content of mapPath in node as a map
+func GetMap(node *yaml.RNode, filePath string, mapPath []string) (map[string]string, error) {
+	mapNode, err := node.Pipe(yaml.Lookup(mapPath...))
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to get %s", mapName)
+		return nil, errors.Wrapf(err, "failed to get %v", mapPath)
 	}
 	m := map[string]string{}
-	if metadata == nil {
+	if mapNode == nil {
 		return m, nil
 	}
-	err = metadata.VisitFields(func(node *yaml.MapNode) error {
+	err = mapNode.VisitFields(func(node *yaml.MapNode) error {
 		v := ""
 		k, err := node.Key.String()
 		if err != nil {
-			return errors.Wrapf(err, "failed to find %s key for path %s", mapName, path)
+			return errors.Wrapf(err, "failed to find %v key for path %s", mapPath, filePath)
 		}
 		v, err = node.Value.String()
 		if err != nil {
-			return errors.Wrapf(err, "failed to find %s %s value for path %s", mapName, k, path)
+			return errors.Wrapf(err, "failed to find %v %s value for path %s", mapPath, k, filePath)
 		}
 		m[strings.TrimSpace(k)] = strings.TrimSpace(v)
 		return nil
 	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get %v", mapPath)
+	}
 	return m, nil
 }
 
-// GetLabels returns the labels for the given file
```

Comment:
Instead of removing the public functions, may be we can modify them to call get map with a string slice containing metadata and annotations for getannotations ?

Reply:
We could. But since the only use was the ones I changed I don't see the point of doing it.

Reply:
Ok as long as we are the only consumers of jx-helpers, it's fine.